pkgs/tools/compression/lz4/default.nix

```
@@ -38,6 +38,7 @@ stdenv.mkDerivation rec {
   ]
     # TODO make full dictionary
     ++ lib.optional stdenv.hostPlatform.isMinGW "TARGET_OS=MINGW"
+    ++ lib.optional (stdenv.buildPlatform != stdenv.hostPlatform && stdenv.hostPlatform.isLinux) "TARGET_OS=Linux"
```

Why special case cross compilation, instead of this simpler version?

```suggestion
    ++ lib.optional stdenv.hostPlatform.isLinux "TARGET_OS=Linux"
```

---

Two reasons:
- It avoids a rebuild which is nice if we wanna backport it
- I feel like it's better to let the upstream build system do its thing unless really necessary, because that's usually more robust to changes

Both of these are minor though, so I'm happy to change it if you feel like they don't matter.

---

> It avoids a rebuild which is nice if we wanna backport it

We have staging release branches. In general we should favour long term maintainability over whether we get to skip a backport.  (If you need it in a hurry, it's okay to do a no-rebuild version, but you should then PR staging to turn it into a rebuild.)

> I feel like it's better to let the upstream build system do its thing unless really necessary, because that's usually more robust to changes

IME it's more robust to have cross builds as close to native as possible, because it limits the scope for something to go wrong with the cross build, but not be noticed because it doesn't have as many users and isn't built on Hydra.

---

Done. Was considering adding `Darwin` as well but it's probably not necessary since atm we can't target Darwin from Linux right?

---

Correct, we can't cross compile to Darwin from any other OS.  Does cross compilation between Darwin architectures work?

---

Surprisingly, this actually works (tested with `nix build .#legacyPackages.x86_64-darwin.pkgsCross.aarch64-darwin.lz4`). It doesn't work the other way around though (`nix build .#legacyPackages.aarch64-darwin.pkgsCross.x86_64-darwin.lz4`). But this is unnecessary anyway with Rosetta `x86_64` emulation.